docs/topics/basic-syntax.md

```diff
@@ -133,44 +133,87 @@ See [Functions](functions.md).
 
 ## Variables
 
-Read-only local variables are defined using the keyword `val`. They can be assigned a value only once.
+In Kotlin, you declare a variable starting with a keyword, `val` or `var`, followed by the name of the variable.
+
+Use the `val` keyword to declare variables that are assigned a value only once. These are read-only local variables that can’t be reassigned a different value
+after initialization: 
 
 ```kotlin
 fun main() {
 //sampleStart
-    val a: Int = 1  // immediate assignment
-    val b = 2   // `Int` type is inferred
-    val c: Int  // Type required when no initializer is provided
-    c = 3       // deferred assignment
+    // Declares the variable x and initializes it with the value of 5
+    val x: Int = 5
+    // 5
 //sampleEnd
-    println("a = $a, b = $b, c = $c")
+    println(x)
 }
 ```
 {kotlin-runnable="true" kotlin-min-compiler-version="1.3" id="kotlin-basic-syntax-val"}
 
-Variables that can be reassigned use the `var` keyword.
+Use the `var` keyword to declare variables that can be reassigned. These are mutable variables, and you can assign other values to them after initialization:
 
 ```kotlin
 fun main() {
 //sampleStart
-    var x = 5 // `Int` type is inferred
+    // Declares the variable x and initializes it with the value of 5
+    var x: Int = 5
+    // Reassigns a new value of 6 to the variable x
     x += 1
+    // 6
 //sampleEnd
-    println("x = $x")
+    println(x)
 }
 ```
 {kotlin-runnable="true" kotlin-min-compiler-version="1.3" id="kotlin-basic-syntax-var"}
 
-You can declare variables at the top level.
+Kotlin supports type inference and automatically identifies the data type of a declared variable. When declaring a variable, you can omit the type after the variable name:
+
+```kotlin
+fun main() {
+//sampleStart
+    // Declares the variable x with the value of 5;`Int` type is inferred
+    val x = 5
+    // 5
+//sampleEnd
+    println(x)
+}
+```
+{kotlin-runnable="true" kotlin-min-compiler-version="1.3" id="kotlin-basic-syntax-inference"}
+
+You can use variables only after initializing them. You can either initialize a variable at the moment of declaration or declare a variable first and initialize it later. 
+In the second case, you must specify the data type:
+
+```kotlin
+fun main() {
+//sampleStart
+    // Initializes the variable x at the moment of declaration; type is not required
+    val x = 5
+    // Declares the variable c without initialization; type is required
+    val c: Int
+    // Initializes the variable c after declaration 
+    c = 3
+    // 5 
+    // 3
+//sampleEnd
+    println(x)
+    println(c)
+}
+```
+{kotlin-runnable="true" kotlin-min-compiler-version="1.3" id="kotlin-basic-syntax-initialize"}
```

Comment:
no suggestion here - but very nice thought of adding the min-compiler-version!! :D